File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/dialogs/HelpDialogBuilder.java
package ch.rmy.android.http_shortcuts.dialogs;

import android.content.Context;
import android.support.annotation.StringRes;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;

import com.afollestad.materialdialogs.MaterialDialog;

import butterknife.Bind;
import butterknife.ButterKnife;
import ch.rmy.android.http_shortcuts.R;
import ch.rmy.android.http_shortcuts.utils.HTMLUtil;

public class HelpDialogBuilder {

    private final MaterialDialog.Builder builder;

    @Bind(R.id.help_text)
    TextView text;

    private View view;

    public HelpDialogBuilder(Context context) {
        LayoutInflater layoutInflater = LayoutInflater.from(context);
        view = layoutInflater.inflate(R.layout.help_dialog, null);
        ButterKnife.bind(this, view);

        builder = new MaterialDialog.Builder(context)
                .customView(view, false)
                .positiveText(android.R.string.ok);
    }

    public HelpDialogBuilder title(@StringRes int title) {
        builder.title(title);
        return this;
    }

    public HelpDialogBuilder message(@StringRes int message) {
        text.setText(HTMLUtil.getHTML(view.getContext(), message));
        return this;
    }

    public HelpDialog build() {
        return new HelpDialog(builder.build());
    }

}

File: HTTPShortcuts/app/src/main/kotlin/ch/rmy/android/http_shortcuts/dialogs/HelpDialogBuilder.kt
package ch.rmy.android.http_shortcuts.dialogs

import android.content.Context
import android.text.method.LinkMovementMethod
import android.view.LayoutInflater
import android.view.View
import android.widget.TextView
import ch.rmy.android.http_shortcuts.R
import ch.rmy.android.http_shortcuts.utils.HTMLUtil

class HelpDialogBuilder(context: Context) : DialogBuilder(context) {

    private val view: View

    init {
        val layoutInflater = LayoutInflater.from(context)
        view = layoutInflater.inflate(R.layout.help_dialog, null)
        view(view)
        positive(android.R.string.ok)
    }

    override fun message(text: CharSequence): DialogBuilder = also {
        val textView = view.findViewById<TextView>(R.id.help_text)
        textView.text = HTMLUtil.getHTML(text.toString())
        textView.movementMethod = LinkMovementMethod.getInstance()
    }

}
